internal/core/src/index/IndexFactory.cpp

```diff
@@ -95,8 +95,26 @@ IndexFactory::CreateVectorIndex(
     }
 
     // create mem index
-    return std::make_unique<VectorMemIndex>(
-        index_type, metric_type, version, file_manager_context);
+    else {
+        switch (data_type) {
+            case DataType::VECTOR_FLOAT: {
+                return std::make_unique<VectorMemIndex<float>>(
+                    index_type, metric_type, version, file_manager_context);
+            }
+            case DataType::VECTOR_BINARY: {
+                return std::make_unique<VectorMemIndex<uint8_t>>(
+                    index_type, metric_type, version, file_manager_context);
+            }
+            default:
```

Comment:
Should we ignore float16 vector?

Reply:
I will submit the part of float16 vector in the next PR, float16 seems less important now.